src/values/length.rs

```diff
@@ -35,6 +35,100 @@ impl LengthPercentage {
   }
 }
 
+/// Either a [`<integer>`](https://www.w3.org/TR/css-values-4/#integers), or the `auto` keyword.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(
+  feature = "serde",
+  derive(serde::Serialize, serde::Deserialize),
+  serde(tag = "type", content = "value", rename_all = "kebab-case")
+)]
+pub enum IntegerOrAuto {
+  /// The `auto` keyword.
+  Auto,
+  /// A `<integer>` value.
+  Integer(CSSInteger),
+}
+
+impl Default for IntegerOrAuto {
+  fn default() -> Self {
+    IntegerOrAuto::Auto
+  }
+}
+
+impl<'i> Parse<'i> for IntegerOrAuto {
+  fn parse<'t>(input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i, ParserError<'i>>> {
+    if input.try_parse(|i| i.expect_ident_matching("auto")).is_ok() {
+      return Ok(IntegerOrAuto::Auto);
+    }
+
+    if let Ok(integer) = input.try_parse(CSSInteger::parse) {
+      return Ok(IntegerOrAuto::Integer(integer));
+    }
+
+    Err(input.new_error_for_next_token())
+  }
+}
+
+impl ToCss for IntegerOrAuto {
+  fn to_css<W>(&self, dest: &mut Printer<W>) -> Result<(), PrinterError>
+  where
+    W: std::fmt::Write,
+  {
+    use IntegerOrAuto::*;
+    match self {
+      Auto => dest.write_str("auto"),
+      Integer(integer) => integer.to_css(dest),
+    }
+  }
+}
+
+/// Either a [`<length>`](https://www.w3.org/TR/css-values-4/#lengths), or the `auto` keyword.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(
+  feature = "serde",
+  derive(serde::Serialize, serde::Deserialize),
+  serde(tag = "type", content = "value", rename_all = "kebab-case")
+)]
+pub enum LengthOrAuto {
```

Comment:
We could probably make a generic type that allows "auto" or a type parameter. Happy to implement that separately.